### ISSUE TITLE
Allow additional Ad Blocker filters

### ISSUE THREAD

**Issue by maverick74** [ghostery/ghostery-extension#22]
Not sure how this would work, but i think it would be good to allow the inclusion of other Filters beside the traditional EasyList.

**Comment by trickpattyFH20** [comment]
@maverick74 Ghostery actually does not use EasyList. We have documentation on the [tracker db here](https://github.com/ghostery/ghostery-extension/blob/master/databases/README.md) If you would like to integrate EasyList or other filter lists with Ghostery you could create a db following that schema and make a build of Ghostery with the new db

**Comment by christophertino** [comment]
This would need to be implemented in the [ad blocker](https://github.com/cliqz-oss/adblocker). 

@zhonghao-cliqz

**Comment by maverick74** [comment]
@trickpattyFH20 But, from what i understand, that would mean i would have "my own Ghostery", since i would have to make a build of it.
(or did i misunderstood your idea?)

My idea was to have this feature in the upstream real Ghostery, allowing users to choose lists they would like to use easily

**Comment by christophertino** [comment]
@maverick74 This is definitely something we will consider as we build out features for the Ad Blocker.  Right now it's just an on/off state. You can see the lists we're implementing [here](https://github.com/cliqz-oss/adblocker/tree/master/assets). I agree it would be nice to have some granularity and choice with the filter lists.

**Comment by maverick74** [comment]
@christophertino great!

It's not that i want to disable any. In fact, my intention is to be able to enable more lists since i occasionally see one or two ads here and there that are blocked on ublock, for example.

**Comment by remusao** [comment]
Hi @maverick74,

If you see some ads, would you maybe consider sharing the websites with us? That would be of great help to either make sure it's not a bug, or consider adding extra list to broaden our coverage.

Thank you so much!

**Comment by maverick74** [comment]
Hi @remusao ,

I'm trying Ghostery with _full block / block all_ in order to compare it with other blockers.

I'll keep a list any webpage URL that has ads and will share them here... if i get any :)

(with this settings, so far, i'm pretty happy)

**Comment by maverick74** [comment]
@christophertino will there be any beta of v8.4 for us to test?

**Comment by remusao** [comment]
@maverick74 It's true that giving the possibility to select custom lists would add some value for some users. On the other hand, Ghostery already gives quite some flexibility in terms of which trackers to block, etc. And in many ways this mechanism is more granular than selecting subscriptions for the adblocker.

Instead, there is another way with which we're playing right now: there could be a switch to ask the adblocker to be more "strict" and also block some tracking as well as ads (but again, this is already something done by Ghostery so the impact would not be as visible as one might think). This would effectively enable some extra subscriptions such as Easyprivacy. On the other hand, we would make sure that the set of rules used play well with the Ghostery experience and limits breakage as much as possible. We really want a "work out of the box" experience for the users and the default settings should work for most people.

**Comment by maverick74** [comment]
@remusao So there is no plans to allow [manually] adding extra filters, for the time being?

analyzing it, adding the possibility of customization (beside blocking capabilities) might give the "psychological" impact on users' mind that they are free to customize / add any extra rules and so it may also help in spreading the product.

On the other way, this would probably only affect "power users" - that may influence others, but that are still are a small group - and this brings in your point that the most important is a great "out of the box" experience.

All in all, thanks for sharing the "other way with which you're playing right now". It seems a good improvement about privacy. Users these days tend to like "switch" :)

**Comment by maverick74** [comment]
update: I've found an example where ublock (with all blocks/lists enabled) is blocking side ads that ghostery is not:
https://www.thewatchcartoononline.tv/

**Comment by remusao** [comment]
@maverick74 Thanks for the report, I can indeed reproduce this issue but it seems to be fixed in the most recent version of the adblocker (tested in latest browser-core and same default subscriptions). So it will definitely be fixed in next release and that does not appear to be an issue with the lists.

**Comment by christophertino** [comment]
@maverick74 related to this topic, we'll be adding the ability to create custom _tracker_ patterns in one of our next releases (~v8.4.5). It's not the same as ad blocker filter lists but it will allow you to block additional URL patterns not currently in the Ghostery DB.

**Comment by maverick74** [comment]
@christophertino It was not exactly what i was hopping for, but will wait and see how it works. :)

Is there any particular reason to discard ad blocker filter lists and use URL patterns instead?

**Comment by christophertino** [comment]
We discussed internally and decided that for the first iteration of this feature we'll concentrate on tracker pattern URLs, since that has always been the primary focus of Ghostery. It's also been a frequent request from people who use Ghostery for web-dev and site-audits. But that's not to say we won't also look at custom filter lists for the Ad-Blocker. I've added that feature as a Discovery option for this ticket.

**Comment by grgnday** [comment]
Ghostery is really one step away from being the main extension to be downloaded for privacy by not having the functionality to add more filters. Some of the most important use cases include language specific filters (both AdGuard and Easylist have them) and annoyance filters. I understand this would require resources to be able to support different syntax (see AdGuard filters) but in the end this is the one thing that keeps me and many that I know from using Ghostery

**Comment by remusao** [comment]
Hi @grgnday,

Thanks for the great feedback. I work on Ghostery's [adblocker](https://github.com/cliqz-oss/adblocker) and would like to give a bit of insight about supporting extra lists. We've been discussing this topic multiple times internally in the past and there are a few important things to consider:

- There is technically nothing that prevents us from enabling more lists in the extension (in fact, syntax from Easylist, uBlock Origin and most of Adguard's is already supported). What needs to be considered though, is how it fits into the Ghostery experience and what value it would add for users (e.g.: more privacy, more comfort, etc.).
- Ghostery already enables language-specific filters *automatically* based on your browsing habits. When the adblocker starts, it checks locally the language of the sites you visit most (it goes without saying that no data ever leaves your browser), then can decide to enable extra filters based on this information. For example if you visit German websites regularly, then [Easylist Germany](https://github.com/easylist/easylistgermany) will be enabled transparently to customize adblocking. We always try to provide the best settings out of the box so that users are as safe as possible, even without the need for manual customization. Automatically enabling language specific lists is one way we achieve this.
- In the light of the previous point, it would be interesting to experiment with extra options such as "enable anti-annoyance mode". If such an option were to be provided, would it alleviate the need to add custom lists manually? Again, more options does not necessarily means better privacy protection or better adblocking; which is why we carefully consider the implications and benefits of adding new features in Ghostery.
- Last but not least, not allowing arbitrary lists in Ghostery at the moment allows us to make sure features behave optimally and do not conflict with each other. For example, the blocking lists used by default in the adblocker are selected and optimized so that they play well with our AI anti-tracking technology. We are also able to pre-bundle them and ship/update them very efficiently to/in the client instead of downloading huge assets which need to be parsed in the client and would result in higher data and CPU usage [1].

Thanks again for your feedback, we will definitely keep the door open for more customization in the future. But it's important to understand that sometimes less is more and that adding features does not necessarily equates with better protection.

Looking forward to read more feedback from the community on this!

Best,
Rémi

---

[1] in a [recent performance study](https://whotracks.me/blog/adblockers_performance_study.html) we showed how Ghostery's adblocker can be initialized *two to three orders of magnitude faster* as compared to other popular adblockers using these techniques (it also uses much less memory for the same number of blocking rules).

**Comment by grgnday** [comment]
I must admit. I should have checked if Ghostery would enable language specific filters before commenting something that you have probably seen multiple times now :) and I really want to thank you again for answering with such an informative manner. One better informed suggestion then, maybe also consider enabling AdGuard’s language specific filters too if not already. They are just like easylist’s filters but it offers filters for more languages such as Turkish, Russian etc. and they should be pretty optimized.

**Comment by maverick74** [comment]
I was also not aware of this "automatic list enabler" feature. That is very nice!!! Maybe that could be a "selling point":

Ghostery Automatically manages your adblocking lists based on your usage so that you don't have to do it yourself.

As for the "enable anti-annoyance mode", being a fan of "the more options, the better" philosophy I would welcome the option, however... I have came to learn - the hard way - that to most people too many options could be a problem.

(eg: i was totally anti-GNOME, for example, and totally pro-KDE! But no! Gnome guys are really right in many things. They're just not "MY" desktop, but for most people, KDE Plasma remains totally unexplored and to some even confusing... So, I get the point of too many options may not be as good As it sounds...)

Anyway, thanks for the explanation. That's one thing I really like about you guys - you do reply your users :)

**Comment by grgnday** [comment]
My suggestion would be to use AdGuard Language filters anywhere EasyList is not available as a fallback filter so that more languages would be covered by Ghostery.

**Comment by arifpedia** [comment]
![Cuplikan layar 2024-08-13 230127](https://github.com/user-attachments/assets/4cbdb385-3209-49d7-8780-d85ef7ff52c7)

Thank you for adding the regional filters feature, but where is the filter for Indonesia? I'm sure the team might have missed it. You can view the list here: 

> [ABPindo](https://raw.githubusercontent.com/heradhis/indonesianadblockrules/master/subscriptions/abpindo.txt)

For additional supplementary filter lists and EasyList variants, you can check this page: [EasyList](https://easylist.to/pages/other-supplementary-filter-lists-and-easylist-variants.html).
